### PR TITLE
configcore,tests: use daemon-reexec to apply watchdog config

### PR DESCRIPTION
**Pull request by pedronis**
Apparently systemd will not apply them with just daemon-reload.

This adds Systemd.DaemonReexec.

### CODE REVIEW

**Comment by pedronis** [comment]
Based on #8435 and #8434.

This will to be tests on a system with an actual hardware watchdog.

File: systemd/systemd.go [hunk 1]
```diff
@@ -291,6 +295,17 @@ func (s *systemd) daemonReloadNoLock() error {
 	return err
 }
 
+func (s *systemd) DaemonReexec() error {
+	if s.mode == GlobalUserMode {
+		panic("cannot call daemon-reexec with GlobalUserMode")
```
Comment (anonymouse64) [inline systemd/systemd.go:300]: maybe a comment about why we can't use daemon-reexec with GlobalUserMode?

**Comment by pedronis** [comment]
the spread test needs fixing for core20 systemd

File: tests/core/watchdog/task.yaml [hunk 2]
```diff
@@ -13,14 +13,20 @@ restore: |
     rm -f "$WATCHDOG_FILE"
 
 execute: |
+    systemd_ver="$(systemctl --version|head -1|cut -d ' ' -f2)"
+    REBOOT_WATCHDOG_PROP=RebootWatchdogUSec
+    if [ "${systemd_ver}" -lt 243 ]; then
+        REBOOT_WATCHDOG_PROP=ShutdownWatchdogUSec
```
Comment (stolowski) [inline tests/core/watchdog/task.yaml:19]: Oh well...

**Comment by pedronis** [comment]
we need to decide the timing for landing this

**Comment by pedronis** [comment]
closing/reopening this

**Pull request merged**